solr/solr-ref-guide/src/morelikethis.adoc

```diff
@@ -16,97 +16,624 @@
+
+==== Search Component Parameters
+
+The MoreLikeThis search component supports the following parameters in addition to the <<Common Handler and Component Parameters,common parameters>> above.
 
 `mlt`::
++
+[%autowidth,frame=none]
+|===
+|Optional |Default: none
+|===
++
 If set to `true`, activates the `MoreLikeThis` component and enables Solr to return `MoreLikeThis` results.
 
 `mlt.count`::
-Specifies the number of similar documents to be returned for each result. The default value is 5.
++
+[%autowidth,frame=none]
+|===
+|Optional |Default: `5`
+|===
++
+Specifies the number of similar documents to be returned for each result.
 
-`mlt.interestingTerms`:: _Same as defined below for the MLT Handler._
+==== Search Component Query and Response
 
-== Parameters for the MoreLikeThisHandler
+The response when using MoreLikeThis as a search component is different than when using the request handler.
 
-The table below summarizes parameters accessible through the `MoreLikeThisHandler`. It supports faceting, paging, and filtering using common query parameters, but does not work well with alternate query parsers.
+In this case, we are using the `/select` request handler and performing a regular query (`q=author:martin`).
+We've asked for MoreLikeThis to be added to the response (`mlt=true`), but otherwise the parameters are the same as the earlier example (we've asked for interesting terms and set minimum term and document frequencies to `0`).
 
-`mlt.match.include`::
-Specifies whether or not the response should include the matched document. If set to false, the response will look like a normal select response.
+[source,bash]
+http://localhost:8983/solr/gettingstarted/select?mlt.fl=name&mlt.mindf=0&mlt.mintf=0&mlt=true&q=author%3Amartin
 
-`mlt.match.offset`::
-Specifies an offset into the main query search results to locate the document on which the `MoreLikeThis` query should operate. By default, the query operates on the first result for the q parameter.
+The response includes the results of our query, in this case 3 documents which have the term "martin" in the author field.
+We've changed the field, however, to find documents that are similar to these based on values in the `name` field (`mlt.fl=name`).
 
-`mlt.interestingTerms`::
-Controls how the `MoreLikeThis` component presents the "interesting" terms (the top TF/IDF terms) for the query.
-It supports three settings:
-The setting `list` lists the terms.
-The setting `none` lists no terms.
-The setting `details` lists the terms along with the boost value used for each term.
-Unless `mlt.boost=true`, all terms will have `boost=1.0`.
+In the response, a `moreLikeThis` section has been added.
+For each document in the results that match our query, a list of document IDs is returned with score values.
+Each of these documents are similar to the document in the result list to varying degrees.
 
+[source,json]
```

Comment:
@alessandrobenedetti I pinged you for review of this PR because this example query originally included the parameter `mlt.interestingTerms` which was supposedly added to the search component in SOLR-12304. However, the response never contained a section for interesting terms - I wasn't sure if there was something I should know to make them appear that was not documented? Or is it perhaps broken since 8.2? It seems that the test still passes (I didn't try it), so wanted to check with you before filing a new jira issue.

Reply:
Hi Cassandra, I am currently way, but back next week, took a note, I'll think about it and keep you updated early next week!

Reply:
Hi Cassandra,
I spent some time reviewing SOLR-12304 issue and the code itself.
It seems the code is ok.
So I tried using one of the latest versions of Solr  (8.8.0) and the parameter seems to work as expected:
`http://localhost:8983/solr/books/select?q=*%3A*&fl=title,subjects&mlt=true&mlt.fl=subjects&mlt.interestingTerms=details`
(an alternative is using mlt.interestingTerms=list)

The Solr response has the interestingTerms section:
```
...
"interestingTerms":{
    "1342":{
      "subjects:fiction":1.0},
    "1952":{
      "subjects:fiction":1.0,
      "subjects:women":1.0,
      "subjects:fiction":1.0},
    "11":{
      "subjects:fiction":1.0,
      "subjects:women":1.0,
      "subjects:fiction":1.0},
    ...
```

Reply:
So, given that, I am not sure I am getting your question, can you elaborate a bit?

Reply:
Sorry, it was my turn to be out for a couple of days.

I'm using 8.8.1 with the request that's shown in this PR:
`http://localhost:8983/solr/gettingstarted/select?mlt.fl=name&mlt.mindf=0&mlt.mintf=0&mlt=true&q=author%3Amartin`

If I add `mlt.interestingTerms=details` to that request I get the exact same output shown here, without an `interestingTerms` section.

Is it possibly related to the field configuration? I indexed the exampledocs shipped with Solr for this, and the `name` field I'm using is stored but does not have termVectors enabled (despite the docs saying it should).

When I change the query to `q=*:*`, I also do not get the interesting terms. And I tried taking off the `mindf` and `mintf` params and still don't get the interesting terms section.

Reply:
So I may have a clue - if I use the techproducts example, which is single-core, it works. My usual mode is SolCloud (a mini-cluster local to my machine) and that seems to be the difference. Should I file a Jira issue for this?

Reply:
Hi @ctargett I think you hit the point!
Reading [](https://issues.apache.org/jira/browse/SOLR-12304) it states explicitly:

> N.B. out of scope for this issue is the distributed case, which is much more complicated and requires much deeper investigations

And the old ticket you found seems to describe the issue quite well!

Reply:
OK, thanks for confirming. I guess I either forgot or didn't fully understand the comment in the issue, but now that I know it was very clear! I'll update the docs to be explicit about this limitation for interesting terms. Thanks for your help!